controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -229,6 +231,30 @@ func (in *KubeadmControlPlane) validateCommon() (allErrs field.ErrorList) {
 	return allErrs
 }
 
+func (in *KubeadmControlPlane) validateEtcd() (allErrs field.ErrorList) {
+
+	if in.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.External != nil && in.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.Local != nil {
+		allErrs = append(
+			allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "kubeadmConfigSpec", "clusterConfiguration", "etcd", "external"),
+				"cannot have both local and external etcd at the same time",
+			),
+		)
+	}
+
+	if in.Spec.KubeadmConfigSpec.InitConfiguration.Etcd.External != nil && in.Spec.KubeadmConfigSpec.InitConfiguration.Etcd.Local != nil {
```

Comment:
not blocking for this PR, but the "real" kubeadm configuration/v1alpha3 does not have embeds ClusterConfiguration; wondering why we are doing this in the bootstrap provider...

Reply:
PR: https://github.com/kubernetes-sigs/cluster-api/pull/2605

Reply:
This case can be removed as well as the first two cases will cover all the checks we need. 

The only reason this check would ever get triggered is if someone first deletes the local configuration and then tries to update both local and external at the same time. I don't think we need to handle this edge case right now.